src/Utils/samples.jl

```diff
@@ -43,32 +42,29 @@ vector).
 
 See the documentation of the respective `Sampler`.
 """
-function sample(X::LazySet{N}, num_samples::Int;
-                sampler=_default_sampler(X),
+function sample(X::LazySet{N}, num_samples::Int, sampler::Type{<:Sampler}=_default_sampler(X);
                 rng::AbstractRNG=GLOBAL_RNG,
                 seed::Union{Int, Nothing}=nothing,
                 VN=Vector{N}) where {N}
-    @assert isbounded(X) "this function requires that the set `X` is bounded"
 
-    D = Vector{VN}(undef, num_samples) # preallocate output
-    _sample!(D, sampler(X); rng=rng, seed=seed)
-    return D
+    S = sampler(X)
+    return sample(X, num_samples, S, rng=rng, seed=seed, VN=VN)
 end
 
-# without argument, returns a single element (instead of a singleton)
-function sample(X::LazySet{N}; kwargs...) where {N}
-    return sample(X, 1; kwargs...)[1]
-end
+function sample(X::LazySet{N}, num_samples::Int, sampler::Sampler;
```

Comment:
chaning the `sampler` from kwarg to positional arg is needed because we have to pass sampler instances -- passing the sampler options outside the struct would be very weird.

in fact, having the set `X` wrapped the inside the sampler is not required; i would rather rewrite the file such that we use multiple dispatch on `sample!(D, X::LazySet, ::Sampler; kwargs...)`.

Reply:
@schillic if you approve the latter idea, or need more clarification, i can address it in a separate PR.

Reply:
Sorry, I don't understand. We can discuss if you want.

Reply:
for example, [here](https://github.com/JuliaReach/LazySets.jl/pull/2547/files#diff-4f0f2a8350175b308a0e1c3aa244dbbcb2ea5495113527869e87127c5afaf57eR25) the argument is the name of the struct (the type) `sampler=LazySets.PolytopeSampler`, but if we want to pass algorithm-specific options, we should be able to pass an *instance* of the struct.

Reply:
I see, sure, go ahead :+1: